.github/workflows/build-test-publish-images.yml

```diff
@@ -50,6 +50,9 @@ jobs:
       - uses: actions/checkout@v4
       - name: Run pre-commit
         uses: pre-commit/action@v3.0.1
+      - name: Run hadolint
+        run: |
+          ci/lint-dockerfiles.sh
```

Comment:
It looks like `hadolint` has pre-commit hooks. I'm curious why you're not using them directly?

https://github.com/hadolint/hadolint/blob/master/.pre-commit-hooks.yaml

Reply:
I mentioned this directly in the PR description.

> ### Why not use `pre-commit`?
>
> _As @KyleFromNVIDIA pointed out to me a while ago, the install hooks for `hadolint`'s `pre-commit` hook are broken: https://github.com/hadolint/hadolint/issues/886_
>
> _We don't change these Dockefiles **that** often... I think the slight extra friction of having to run a script manually or just get feedback from CI should not block us from adopting this tool._

Because of that issue, you'd have to manually install `hadolint` to run `pre-commit run` locally... I don't think the friction is worth it.

Reply:
Gah sorry I totally missed that section. Makes sense!

Reply:
ha no problem